selfdrive/camerad/main.cc

```diff
@@ -1062,6 +1167,32 @@ void init_buffers(VisionState *s) {
     assert(err == 0);
   }
 
+  s->prg_debayer_rear = build_debayer_program(s, s->cameras.rear.ci.frame_width, s->cameras.rear.ci.frame_height,
+                                                   s->cameras.rear.ci.frame_stride,
+                                                 s->rgb_width, s->rgb_height, s->rgb_stride,
+                                                 s->cameras.rear.ci.bayer_flip, s->cameras.rear.ci.hdr);
+  s->krnl_debayer_rear = clCreateKernel(s->prg_debayer_rear, "debayer10", &err);
+  assert(err == 0);
+
+  s->prg_rgb_laplacian = build_conv_program(s, s->rgb_width/NUM_SEGMENTS_X, s->rgb_height/NUM_SEGMENTS_Y, 
+                                            3);
+  s->krnl_rgb_laplacian = clCreateKernel(s->prg_rgb_laplacian, "rgb2gray_conv2d", &err);
```

Comment:
Where are these kernels and buffers destroyed?

Reply:
I think they are destroyed when cl context is released